### ISSUE TITLE
Error resolving query due to concatenation assignment

### ISSUE THREAD

**Issue by hemberger** [staabm/phpstan-dba#541]
If I construct a query by first assigning a literal string, and then using concatenation assignment to append another literal string, e.g.

```php
$query = 'SELECT * FROM account WHERE account_id = :account_id';
$query .= ' AND referral_id = :referral_id';
\PHPStan\dumpType($query);
$db->preparedQuery($query, ['account_id' => 1, 'referral_id' => 3]);
```
then PHPStan reports the following errors:
```
3   Dumped type: 'SELECT * FROM account WHERE account_id = :account_id AND referral_id = :referral_id'
4   Value :referral_id is given, but the query does not contain this placeholder.
```
As we can see, the type is correctly inferred as the concatenation of the two literal strings, but something went wrong with resolving the query string - it's missing the RHS of the concatenation assignment.

Note that the issue does not apply to normal concatenation, only concatenation assignment.

I dug a little deeper and found that the Expr gets chopped off somewhere in the `ExpressionFinder::findAssignmentExpression` at https://github.com/staabm/phpstan-dba/blob/d43520b693e5fb88eeafbb8920638b35854568cf/src/QueryReflection/QueryReflection.php#L312

Prior to this line, dumping `$scope->getType($queryExpr)` gives:
```
object(PHPStan\Type\Constant\ConstantStringType)#12139 (4) {
  ["objectType":"PHPStan\Type\Constant\ConstantStringType":private]=>
  NULL
  ["arrayKeyType":"PHPStan\Type\Constant\ConstantStringType":private]=>
  NULL
  ["value":"PHPStan\Type\Constant\ConstantStringType":private]=>
  string(83) "SELECT * FROM account WHERE account_id = :account_id AND referral_id = :referral_id"
  ["isClassString":"PHPStan\Type\Constant\ConstantStringType":private]=>
  bool(false)
}
```
which we can see has the entire concatenated string in it. However, after this line, dumping `$scope->getType($queryStringExpr)` gives:
```
object(PHPStan\Type\Constant\ConstantStringType)#13250 (4) {
  ["objectType":"PHPStan\Type\Constant\ConstantStringType":private]=>
  NULL
  ["arrayKeyType":"PHPStan\Type\Constant\ConstantStringType":private]=>
  NULL
  ["value":"PHPStan\Type\Constant\ConstantStringType":private]=>
  string(52) "SELECT * FROM account WHERE account_id = :account_id"
  ["isClassString":"PHPStan\Type\Constant\ConstantStringType":private]=>
  bool(false)
}
```
which only has the string from the initial assignment in it.

I tried to figure out what was going on in `ExpressionFinder`, but I'm a bit too unfamiliar with the PhpParser library. Any help would be greatly appreciated. Thank you!

**Comment by staabm** [comment]
thx for the detailed report. I came up with a fix in #542.

please give it a try

**Comment by hemberger** [comment]
Looks to be working now, thanks for the quick fix!

**Comment by staabm** [comment]
thanks. released in 0.2.63
